client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,170 +1,205 @@
+
+      return this.fetchAgentsIds().then((ids) => {
+        this.agents.ids = ids;
+      }).catch((err) => {
+        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_get_error'), _.get(err, 'data.message')].join(' '));
         return this.$q.reject(err);
-      })
-      .finally(() => {
-        _.set(agent, 'isUpdating', false);
       });
-  }
-
-  cancelAddAgent(pos) {
-    if (pos === 0 && this.addAgentForm.numbers.length > 1) {
-      this.addAgentForm.numbers.shift();
-    } else if (this.addAgentForm.numbers.length > 1) {
-      _.pullAt(this.addAgentForm.numbers, pos);
-    } else {
-      this.addAgentForm.numbers[0] = null;
     }
-  }
-
-  addAgents() {
-    const modal = this.$uibModal.open({
-      animation: true,
-      templateUrl: 'app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx-modal.html',
-      controller: 'telecomTelephonyAliasConfigurationAgentsOvhPabxModal',
-      controllerAs: '$ctrl',
-    });
-    modal.result.then(() => {
+
+    orderBy(prop, asc) {
+      this.orderedAscAgents = !this.orderedAscAgents;
+      if (asc != null) {
+        this.orderedAscAgents = asc;
+      }
+
+      this.agents.paginated = _.sortBy(this.agents.paginated, prop);
+
+      if (!this.orderedAscAgents) {
+        this.agents.paginated = this.agents.paginated.reverse();
+      }
+    }
+
+    fetchAgentsIds() {
+      this.agents.isLoading = true;
+      return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
+        .query({
+          billingAccount: this.$stateParams.billingAccount,
+          serviceName: this.$stateParams.serviceName,
+        }).$promise.finally(() => {
+          this.agents.isLoading = false;
+        });
+    }
+
+    fetchAgent(id) {
+      return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
+        .get({
+          billingAccount: this.$stateParams.billingAccount,
+          serviceName: this.$stateParams.serviceName,
+          agentId: id,
+        }).$promise;
+    }
+
+    getSelectedAgentIds() {
+      return _.keys(this.agents.selected);
+    }
+
+    onTransformItemDone(items) {
+      this.$timeout(() => this.orderBy('number', this.orderedAscAgents));
+      return items;
+    }
+
+    deleteAgents() {
+      this.agents.isDeleting = true;
+      return this.$q
+        .all(this.getSelectedAgentIds()
+          .map(id => this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
+            .remove({
+              billingAccount: this.$stateParams.billingAccount,
+              serviceName: this.$stateParams.serviceName,
+              agentId: id,
+            }).$promise.then(() => {
+              _.pull(this.agents.id, parseInt(id, 10));
+            })))
+        .then(() => {
+          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
+          return this.fetchAgentsIds().then((ids) => {
```

Comment:
Better `return this.fetchAgentsIds()` and move the then block into a one at the same level than `catch` and `finally` for readability.

Reply:
Sorry but I don't understand your feedback on this return. Could you explain it to me?

Reply:
Sure I mean to do this, to improve readability

```suggestion
.then(() => {
          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
          return this.fetchAgentsIds();
})
.then((ids) => { 
   // Your code
})
```